src/coreclr/jit/register.h

```diff
@@ -154,4 +154,738 @@ REGDEF(STK,    8+KBASE,    0x0000,       "STK"  )
+#endif // defined(TARGET_LOONGARCH64)
+
+#if defined(TARGET_RISCV64)
+#undef REG_R0
+#define REG_R0 JITREG_R0
+#undef REG_RA
+#define REG_RA JITREG_RA
+#undef REG_SP
+#define REG_SP JITREG_SP
+#undef REG_GP
+#define REG_GP JITREG_GP
+#undef REG_TP
+#define REG_TP JITREG_TP
+#undef REG_T0
+#define REG_T0 JITREG_T0
+#undef REG_T1
+#define REG_T1 JITREG_T1
+#undef REG_T2
+#define REG_T2 JITREG_T2
+#undef REG_FP
+#define REG_FP JITREG_FP
+#undef REG_S1
+#define REG_S1 JITREG_S1
+#undef REG_A0
+#define REG_A0 JITREG_A0
+#undef REG_A1
+#define REG_A1 JITREG_A1
+#undef REG_A2
+#define REG_A2 JITREG_A2
+#undef REG_A3
+#define REG_A3 JITREG_A3
+#undef REG_A4
+#define REG_A4 JITREG_A4
+#undef REG_A5
+#define REG_A5 JITREG_A5
+#undef REG_A6
+#define REG_A6 JITREG_A6
+#undef REG_A7
+#define REG_A7 JITREG_A7
+#undef REG_S2
+#define REG_S2 JITREG_S2
+#undef REG_S3
+#define REG_S3 JITREG_S3
+#undef REG_S4
+#define REG_S4 JITREG_S4
+#undef REG_S5
+#define REG_S5 JITREG_S5
+#undef REG_S6
+#define REG_S6 JITREG_S6
+#undef REG_S7
+#define REG_S7 JITREG_S7
+#undef REG_S8
+#define REG_S8 JITREG_S8
+#undef REG_S9
+#define REG_S9 JITREG_S9
+#undef REG_S10
+#define REG_S10 JITREG_S10
+#undef REG_S11
+#define REG_S11 JITREG_S11
+#undef REG_T3
+#define REG_T3 JITREG_T3
+#undef REG_T4
+#define REG_T4 JITREG_T4
+#undef REG_T5
+#define REG_T5 JITREG_T5
+#undef REG_T6
+#define REG_T6 JITREG_T6
+#undef REG_R8
+#define REG_R8 JITREG_R8
+#undef REG_ZERO
+#define REG_ZERO JITREG_ZERO
+#undef REG_FT0
+#define REG_FT0 JITREG_FT0
+#undef REG_FT1
+#define REG_FT1 JITREG_FT1
+#undef REG_FT2
+#define REG_FT2 JITREG_FT2
+#undef REG_FT3
+#define REG_FT3 JITREG_FT3
+#undef REG_FT4
+#define REG_FT4 JITREG_FT4
+#undef REG_FT5
+#define REG_FT5 JITREG_FT5
+#undef REG_FT6
+#define REG_FT6 JITREG_FT6
+#undef REG_FT7
+#define REG_FT7 JITREG_FT7
+#undef REG_FS0
+#define REG_FS0 JITREG_FS0
+#undef REG_FS1
+#define REG_FS1 JITREG_FS1
+#undef REG_FA0
+#define REG_FA0 JITREG_FA0
+#undef REG_FA1
+#define REG_FA1 JITREG_FA1
+#undef REG_FA2
+#define REG_FA2 JITREG_FA2
+#undef REG_FA3
+#define REG_FA3 JITREG_FA3
+#undef REG_FA4
+#define REG_FA4 JITREG_FA4
+#undef REG_FA5
+#define REG_FA5 JITREG_FA5
+#undef REG_FA6
+#define REG_FA6 JITREG_FA6
+#undef REG_FA7
+#define REG_FA7 JITREG_FA7
+#undef REG_FS2
+#define REG_FS2 JITREG_FS2
+#undef REG_FS3
+#define REG_FS3 JITREG_FS3
+#undef REG_FS4
+#define REG_FS4 JITREG_FS4
+#undef REG_FS5
+#define REG_FS5 JITREG_FS5
+#undef REG_FS6
+#define REG_FS6 JITREG_FS6
+#undef REG_FS7
+#define REG_FS7 JITREG_FS7
+#undef REG_FS8
+#define REG_FS8 JITREG_FS8
+#undef REG_FS9
+#define REG_FS9 JITREG_FS9
+#undef REG_FS10
+#define REG_FS10 JITREG_FS10
+#undef REG_FS11
+#define REG_FS11 JITREG_FS11
+#undef REG_FT8
+#define REG_FT8 JITREG_FT8
+#undef REG_FT9
+#define REG_FT9 JITREG_FT9
+#undef REG_FT10
+#define REG_FT10 JITREG_FT10
+#undef REG_FT11
+#define REG_FT11 JITREG_FT11
+#undef REG_STK
+#define REG_STK JITREG_STK
+#endif // defined(TARGET_RISCV64)
```

Comment:
Do we really need this for all targets? Can we just do it for only the targets with the conflicts?

Reply:
Removed

Reply:
My understanding is that we only need this for x64 and arm32, so a few more can be removed. Also would suggest to self contain the workaround where the enum gets defined if that's possible (so do the undef+defs right after the enums)

Reply:
> My understanding is that we only need this for x64 and arm32

that doesn't match with what I get, still needed for all four

Reply:
It's needed for arm32, x86 and x64 at very least. It may be needed for RISC-V (depends on `__USE_GNU` define). It's not needed for arm64 with the NDK 27 version on my machine but there's no guarantee about different versions.

Reply:
> It's needed for arm32, x86 and x64 at very least. It may be needed for RISC-V (depends on `__USE_GNU` define). It's not needed for arm64 with the NDK 27 version on my machine but there's no guarantee about different versions.

it refuses to build on arm64 if I remove it. It's possible that it's only because of ARM32 subset of registers, but don't think we want to complicate code just redefine them only

Reply:
Are we going to be building x86 JIT for Android? I'm not quite sure I understand how we get into the situation of building that target on Android. Can you explain?

> it refuses to build on arm64 if I remove it.

Hmm, odd. @janvorli pointed out in the issue that he was able to get it to build for arm64. 

> It's possible that it's only because of ARM32 subset of registers, but don't think we want to complicate code just redefine them only

Isn't this already split into its own section in target.h?

Reply:
Is the problem that building the crossjits also conflicts with the host definitions? If so then it makes sense. We could probably skip the builds of the crossjits when targeting Android, I assume they aren't going to be used -- but either way is fine with me.

Reply:
> Are we going to be building x86 JIT for Android?

android-x86 is supported platform both by MonoVM and up-to-date NDK (https://developer.android.com/ndk/guides/abis).

Reply:
> Are we going to be building x86 JIT for Android? I'm not quite sure I understand how we get into the situation of building that target on Android. Can you explain?

Afaik we would need it for x86 Android emulator images, which are still used.

/cc: @akoeplinger

Reply:
I do not think RyuJIT for x86 Android will work without some (potentially significant) investments. So the build fixes could be done as part of those investments if/when we get there. But either way is ok with me -- either way we have workarounds (in JIT sources or in CMake build scripts).

Reply:
> Hmm, odd. @janvorli https://github.com/dotnet/runtime/pull/110471#issuecomment-2602355933 that he was able to get it to build for arm64.

From what I read in that thread it used to build for x64 as well for a different NDK version

Reply:
Anway, if you have ideas how narrow down the amount of undef, please feel free to send a PR once Android support is landed, so that can be validate on CI instead of creating build errors for that PR

Reply:
> I believe linux x86 it is actively being used by Samsung.

I think it was broken for a while and doesn't pass (some of) the tests. Hello World worked last time I tried. I intended to look into it at some point and I would certainly prefer to keep the x86 code around.

Reply:
> Why do you think it would not work? While it is not supported by us, I believe linux x86 it is actively being used by Samsung.

I have not seen contributions to linux-x86 for a long time, and we have recently completed some quite substantial internal ABI representation changes within the JIT. Since it is untested I would be surprised if something hasn't bitrotted.